src/bin/upgrade/main.rs

```diff
@@ -0,0 +1,108 @@
+//! `cargo upgrade`
+
+#![warn(missing_docs, missing_debug_implementations, missing_copy_implementations, trivial_casts,
+        trivial_numeric_casts, unsafe_code, unstable_features, unused_import_braces,
+        unused_qualifications)]
+
+extern crate docopt;
+extern crate pad;
+#[macro_use]
+extern crate serde_derive;
+extern crate toml;
+
+use std::error::Error;
+use std::io::{self, Write};
+use std::process;
+
+extern crate cargo_edit;
+use cargo_edit::{Manifest, get_latest_dependency};
+
+static USAGE: &'static str = r"
+Upgrade all dependencies in a manifest file to the latest version.
+
+Usage:
+    cargo upgrade [--dependency <dep>...] [--manifest-path <path>]
+    cargo upgrade (-h | --help)
+    cargo upgrade (-V | --version)
+
+Options:
+    -d --dependency <dep>       Specific dependency to upgrade. If this option is used, only the
+                                specified dependencies will be upgraded.
+    --manifest-path <path>      Path to the manifest to upgrade.
+    -h --help                   Show this help page.
+    -V --version                Show version.
+
+Dev, build, and all target dependencies will also be upgraded. Only dependencies from crates.io are
+supported. Git/path dependencies will be ignored.
+";
+
+/// Docopts input args.
+#[derive(Debug, Deserialize)]
+struct Args {
+    /// `--dependency -d <dep>`
+    flag_dependency: Vec<String>,
+    /// `--manifest-path <path>`
+    flag_manifest_path: Option<String>,
+    /// `--version`
+    flag_version: bool,
+}
+
+fn is_version_dependency(dep: &toml::Value) -> bool {
+    dep.as_table()
+        .map(|table| {
+            // Not a version dependency if the `git` or `path` keys are present.
+            !(table.contains_key("git") || table.contains_key("path"))
+        })
+        .unwrap_or(true)
```

Comment:
The comment seems redundant as the code says basically the same thing. I also think I'd nowadays write this as 

```rust
if let Some(table) = dep.as_table() {
    !table.contains_key("git") && !table.contains_key("path")
} else {
    true
}
```